### ISSUE TITLE
Quick Start example fails with device error

### ISSUE THREAD

**Issue by ytsaig** [cdpierse/transformers-interpret#16]
Hi,

At least on my environment, with transformers 3.5.0 and pytorch 1.6.0, the quickstart example fails with an error about the device used, see below. I was able to get it to run on cpu by setting `cls_explainer.device = torch.device("cpu")`, but I'm wondering if there's a way to run it on gpu?

```python
import torch
import transformers
```

```python
torch.__version__
```
    '1.6.0'

```python
transformers.__version__
```
    '3.5.0'

```python
from transformers import AutoModelForSequenceClassification, AutoTokenizer
model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
```

```python
from transformers_interpret import SequenceClassificationExplainer
cls_explainer = SequenceClassificationExplainer("I love you, I like you", model, tokenizer)
attributions = cls_explainer()
```

    ---------------------------------------------------------------------------

    RuntimeError                              Traceback (most recent call last)

    <ipython-input-4-b5ac24f5e40f> in <module>
          1 from transformers_interpret import SequenceClassificationExplainer
          2 cls_explainer = SequenceClassificationExplainer("I love you, I like you", model, tokenizer)
    ----> 3 attributions = cls_explainer()
    

    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in __call__(self, text, index, class_name)
        133 
        134     def __call__(self, text: str = None, index: int = None, class_name: str = None):
    --> 135         return self.run(text, index, class_name)
        136 
        137     def __str__(self):

    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in run(self, text, index, class_name)
         51             self.text = text
         52 
    ---> 53         self._calculate_attributions(index=index, class_name=class_name)
         54         return self.attributions
         55 

    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in _calculate_attributions(self, index, class_name)
        114                 self.selected_index = self.predicted_class_index
        115         else:
    --> 116             self.selected_index = self.predicted_class_index
        117 
        118         if self.attribution_type == "lig":

    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in predicted_class_index(self)
         62     def predicted_class_index(self):
         63         if self.input_ids is not None:
    ---> 64             preds = self.model(self.input_ids)[0]
         65             self.pred_class = torch.argmax(torch.softmax(preds, dim=0)[0])
         66             return torch.argmax(torch.softmax(preds, dim=1)[0]).detach().numpy()

    /data/anaconda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
        720             result = self._slow_forward(*input, **kwargs)
        721         else:
    --> 722             result = self.forward(*input, **kwargs)
        723         for hook in itertools.chain(
        724                 _global_forward_hooks.values(),

    /data/anaconda/lib/python3.7/site-packages/transformers/modeling_distilbert.py in forward(self, input_ids, attention_mask, head_mask, inputs_embeds, labels, output_attentions, output_hidden_states, return_dict)
        629             output_attentions=output_attentions,
        630             output_hidden_states=output_hidden_states,
    --> 631             return_dict=return_dict,
        632         )
        633         hidden_state = distilbert_output[0]  # (bs, seq_len, dim)

    /data/anaconda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
        720             result = self._slow_forward(*input, **kwargs)
        721         else:
    --> 722             result = self.forward(*input, **kwargs)
        723         for hook in itertools.chain(
        724                 _global_forward_hooks.values(),

    /data/anaconda/lib/python3.7/site-packages/transformers/modeling_distilbert.py in forward(self, input_ids, attention_mask, head_mask, inputs_embeds, output_attentions, output_hidden_states, return_dict)
        479 
        480         if inputs_embeds is None:
    --> 481             inputs_embeds = self.embeddings(input_ids)  # (bs, seq_length, dim)
        482         return self.transformer(
        483             x=inputs_embeds,

    /data/anaconda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
        720             result = self._slow_forward(*input, **kwargs)
        721         else:
    --> 722             result = self.forward(*input, **kwargs)
        723         for hook in itertools.chain(
        724                 _global_forward_hooks.values(),

    /data/anaconda/lib/python3.7/site-packages/transformers/modeling_distilbert.py in forward(self, input_ids)
        106         position_ids = position_ids.unsqueeze(0).expand_as(input_ids)  # (bs, max_seq_length)
        107 
    --> 108         word_embeddings = self.word_embeddings(input_ids)  # (bs, max_seq_length, dim)
        109         position_embeddings = self.position_embeddings(position_ids)  # (bs, max_seq_length, dim)
        110 

    /data/anaconda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
        720             result = self._slow_forward(*input, **kwargs)
        721         else:
    --> 722             result = self.forward(*input, **kwargs)
        723         for hook in itertools.chain(
        724                 _global_forward_hooks.values(),

    /data/anaconda/lib/python3.7/site-packages/torch/nn/modules/sparse.py in forward(self, input)
        124         return F.embedding(
        125             input, self.weight, self.padding_idx, self.max_norm,
    --> 126             self.norm_type, self.scale_grad_by_freq, self.sparse)
        127 
        128     def extra_repr(self) -> str:

    /data/anaconda/lib/python3.7/site-packages/torch/nn/functional.py in embedding(input, weight, padding_idx, max_norm, norm_type, scale_grad_by_freq, sparse)
       1812         # remove once script supports set_grad_enabled
       1813         _no_grad_embedding_renorm_(weight, input, max_norm, norm_type)
    -> 1814     return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
       1815 
       1816 

    RuntimeError: Expected object of device type cuda but got device type cpu for argument #1 'self' in call to _th_index_select

**Comment by cdpierse** [comment]
Hi @ytsaig , thank you for pointing this out to me. I think the issue was that I wasn't moving the model's params over to GPU when it was available. I've changed that with #17 and just pushed a patch fix to pypi that should hopefully remedy this. 

Could you run `pip install transformers-interpret -U` and try running the quick start again, do let me know If it works. Thanks again.

**Comment by ytsaig** [comment]
Thanks for the super quick response! I tried updating and I now get a different error:

```python
import torch
import transformers
```


```python
torch.__version__
```




    '1.6.0'




```python
transformers.__version__
```




    '3.5.0'




```python
from transformers import AutoModelForSequenceClassification, AutoTokenizer
model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased-finetuned-sst-2-english")
```


```python
from transformers_interpret import SequenceClassificationExplainer
cls_explainer = SequenceClassificationExplainer("I love you, I like you", model, tokenizer)
attributions = cls_explainer()
```


    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-5-b5ac24f5e40f> in <module>
          1 from transformers_interpret import SequenceClassificationExplainer
          2 cls_explainer = SequenceClassificationExplainer("I love you, I like you", model, tokenizer)
    ----> 3 attributions = cls_explainer()
    

    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in __call__(self, text, index, class_name)
        133 
        134     def __call__(self, text: str = None, index: int = None, class_name: str = None):
    --> 135         return self.run(text, index, class_name)
        136 
        137     def __str__(self):


    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in run(self, text, index, class_name)
         51             self.text = text
         52 
    ---> 53         self._calculate_attributions(index=index, class_name=class_name)
         54         return self.attributions
         55 


    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in _calculate_attributions(self, index, class_name)
        114                 self.selected_index = self.predicted_class_index
        115         else:
    --> 116             self.selected_index = self.predicted_class_index
        117 
        118         if self.attribution_type == "lig":


    /data/anaconda/lib/python3.7/site-packages/transformers_interpret/explainers/sequence_classification.py in predicted_class_index(self)
         64             preds = self.model(self.input_ids)[0]
         65             self.pred_class = torch.argmax(torch.softmax(preds, dim=0)[0])
    ---> 66             return torch.argmax(torch.softmax(preds, dim=1)[0]).detach().numpy()
         67 
         68         else:


    TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.



```python

```

**Comment by cdpierse** [comment]
Okay, so it seems like there were a number of issues related to the CPU/GPU compact. Serves me right for only testing on CPU. I've just pushed a number of patches and `v0.1.9` should do the trick now.  

Effectively, once the attributions were calculated and I was moving them to a numpy array I wasn't moving the tensor format CPU first so it was causing the device type error. 

I've tested that example on both a GPU colab instance and CPU instance and all seems good now. 

Thanks again for pointing this out, GPU compatibility is really important to this project as attributions are calculated much much faster with them available, so I appreciate you pointing this all out.
